### PR TITLE
Add ability to configure IPAM config for CreateNetworkCmd 

### PR DESCRIPTION
**Pull request by vuminhkh**
Hi,

I'm fixing a minor issue in network creation API. Please let me know if it's ok for you. Thanks.

Regards,

Minh Khang VU

---
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/550)

### CODE REVIEW

File: src/test/java/com/github/dockerjava/netty/exec/CreateNetworkCmdExecTest.java [hunk 1]
```diff
@@ -40,8 +40,9 @@ public void afterMethod(ITestResult result) {
     public void createNetwork() throws DockerException {
 
         String networkName = "testNetwork";
-
-        CreateNetworkResponse createNetworkResponse = dockerClient.createNetworkCmd().withName(networkName).exec();
+        Network.Ipam.Config config = new Network.Ipam.Config();
+        config.setSubnet("10.67.79.0/24");
+        CreateNetworkResponse createNetworkResponse = dockerClient.createNetworkCmd().withName(networkName).withIpamConfig(config).exec();
```
Comment (KostyaSha) [inline src/test/java/com/github/dockerjava/netty/exec/CreateNetworkCmdExecTest.java]: please create separate test

File: src/main/java/com/github/dockerjava/core/command/CreateNetworkCmdImpl.java [hunk 2]
```diff
@@ -58,6 +58,9 @@ public CreateNetworkCmd withDriver(String driver) {
 
     @Override
     public CreateNetworkCmd withIpamConfig(Ipam.Config config) {
+        if (this.ipam == null) {
+            this.ipam = new Ipam();
+        }
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/command/CreateNetworkCmdImpl.java]: if it null, why do you need set it? Is this field required for API request?

Comment (vuminhkh) [inline src/main/java/com/github/dockerjava/core/command/CreateNetworkCmdImpl.java]: It's an optional config to create a network
https://docs.docker.com/engine/reference/api/docker_remote_api_v1.21/#create-a-network
If I don't set it then the next line will raise NPE, for me it's a bug that prevent user to set custom configuration for network creation.

File: src/test/java/com/github/dockerjava/core/command/CreateNetworkCmdImplTest.java [hunk 3]
```diff
@@ -5,11 +5,7 @@
 import com.github.dockerjava.api.model.Network;
 import com.github.dockerjava.client.AbstractDockerClientTest;
 import org.testng.ITestResult;
-import org.testng.annotations.AfterMethod;
-import org.testng.annotations.AfterTest;
-import org.testng.annotations.BeforeMethod;
-import org.testng.annotations.BeforeTest;
-import org.testng.annotations.Test;
+import org.testng.annotations.*;
```
Comment (KostyaSha) [inline src/test/java/com/github/dockerjava/core/command/CreateNetworkCmdImplTest.java]: please don't use hash imports

**Comment by marcuslinke** [comment]
@vuminhkh Please rename tests to `createNetworkWithIpamConfig`. Thanks! Beside this it LGTM. @KostyaSha WDYT?

File: src/main/java/com/github/dockerjava/core/command/CreateNetworkCmdImpl.java (hunk 2 above)
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/command/CreateNetworkCmdImpl.java]: NPE where? In docker daemon?

**Comment by marcuslinke** [comment]
@vuminhkh Thanks for contributing!

File: src/main/java/com/github/dockerjava/core/command/CreateNetworkCmdImpl.java [hunk 4]
```diff
@@ -58,6 +58,9 @@ public CreateNetworkCmd withDriver(String driver) {
...
         this.ipam.getConfig().add(config);
```
Comment (marcuslinke) [inline src/main/java/com/github/dockerjava/core/command/CreateNetworkCmdImpl.java]: No. In this line.

**Pull request merged**